### PR TITLE
Fixes #34861 - Syncable yum-format repository exports

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by parthaa** [comment]
connected to -> https://github.com/Katello/katello/pull/10183

File: lib/hammer_cli_katello/content_export.rb [hunk 1]
```diff
@@ -44,6 +44,45 @@ def execute
       end
     end
 
+    class GenerateListingCommand < HammerCLIKatello::Command
+      desc _("Generates listing file on each directory of a syncable export. This command "\
+        + "only needs to be used if the export was performed asynchronously or if the listing files"\
+        + " were lost. "\
```
Comment (jeremylenz) [inline lib/hammer_cli_katello/content_export.rb:50]: The most common reason the listing files might be missing is that hammer didn't have permissions..
```suggestion
        + "only needs to be used if the export was performed asynchronously or if the listing files"\
        + "were not generated at time of export. "\
```

File: lib/hammer_cli_katello/content_export_helper.rb [hunk 2]
```diff
@@ -54,6 +67,37 @@ def fetch_export_history_from_task(task)
...
+        paths = Find.find("#{export_history["path"]}/content").select do |path|
+          File.directory?(path) &&
+            ignorables.none? {|ignorable| path.start_with?(ignorable)}
+        end
+
+        paths.each do |dir|
+          directories = Dir.chdir(dir) { Dir['*'] }
+          File.write("#{dir}/listing", directories.join("\n"))
+        end
+        output.print_message _("Generated #{export_history["path"]}")
```
Comment (jeremylenz) [inline lib/hammer_cli_katello/content_export_helper.rb:93]:
```suggestion
        output.print_message _("Generated listing files in #{export_history["path"]}")
```

File: lib/hammer_cli_katello/content_export_helper.rb [hunk 3]
```diff
@@ -54,6 +67,37 @@ def fetch_export_history_from_task(task)
...
+      rescue SystemCallError
+        output.print_message _("Unable to access/write listing files to '#{export_history["path"]}'." \
+                               + "To generate listing files run the command below as a root user ")
+        output.print_message (" hammer content-export generate-listing --id #{export_history['id']}")
```
Comment (jeremylenz) [inline lib/hammer_cli_katello/content_export_helper.rb:97]:
```suggestion
        output.print_message (" hammer content-export generate-listing --id #{export_history['id']}")
        output.print_message _("(note: This command must be run from the same machine as your pulpcore installation")
```

**Review by jeremylenz — COMMENTED** [review]
Thanks for the live demo @parthaa!

Performed a CV version export and confirmed listing files were generated:
```
[vagrant@centos8-katello-devel hammer-cli-katello]$ hammer content-export complete version --format=syncable --content-view-id=22 --version=1.0
[......................................................................................................................................................................] [100%]
Generated /var/lib/pulp/exports/Export_Org/SyncableView/1.0/2022-07-13T20-40-44-00-00
[vagrant@centos8-katello-devel hammer-cli-katello]$ cd /var/lib/pulp/exports/Export_Org/SyncableView/1.0/2022-07-13T20-40-44-00-00
[vagrant@centos8-katello-devel 2022-07-13T20-40-44-00-00]$ ll
total 0
drwxrwxr-x+ 3 pulp pulp 33 Jul 13 20:40 content
[vagrant@centos8-katello-devel 2022-07-13T20-40-44-00-00]$ find .
.
./content
./content/dist
./content/dist/layered
./content/dist/layered/rhel8
./content/dist/layered/rhel8/x86_64
./content/dist/layered/rhel8/x86_64/ansible
./content/dist/layered/rhel8/x86_64/ansible/2.8
./content/dist/layered/rhel8/x86_64/ansible/2.8/os
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.2-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.13-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.1-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.11-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.12-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.6-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/a27718cc28ec6d71432e0ef3e6da544b7f9d93f6bb7d0a55aacd592d03144b70-comps.xml
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/b728eb62aa7e4f6d9ddcef3aee6fade1d89c921323f7f486edb48e629828ccbc-filelists.sqlite.bz2
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/repomd.xml
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/7c580f8bd872b59639aa04280dd3818115ad10c3eb8fea8c1ce924b27ad9c501-filelists.xml.gz
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/3a043c6ae0d22cb4fb861f6d56899bfbb361fc943c7636ea303b399aa4341106-primary.xml.gz
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/d37c42017eefb515ac68f1c14bbe6024d051907690b93a9abf90fee70a54fe58-other.xml.gz
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/50b9a47d-c1b0-4bb1-bc44-1a8c9d9d60f3
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/5d2bf6693fa72826821286a65c1977320c0f75c94440391799a1cc23caef2774-primary.sqlite.bz2
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/ecddb990a73ab03a16976494303a3ba388dffa64fd0aab6bf670e1139c3fd1a0-updateinfo.xml.gz
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/repodata/af5e04be034d62dd035f39b31eb60e1f1b8a26b78f5a91edb9c80a13ea86fdb3-other.sqlite.bz2
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.10-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.18-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.16-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.8-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.3-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.4-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/sshpass-1.06-3.el8ae.x86_64.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.7-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.17-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.5-2.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.0-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/os/ansible-2.8.15-1.el8ae.noarch.rpm
./content/dist/layered/rhel8/x86_64/ansible/2.8/listing
./content/dist/layered/rhel8/x86_64/ansible/listing
./content/dist/layered/rhel8/x86_64/openstack-devtools
./content/dist/layered/rhel8/x86_64/openstack-devtools/15
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/openstack-packstack-puppet-14.0.1-0.20200114160446.96fbb95.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/93f5e08dd6da667f43231918de780b34e9b74e1496c7581028ef2c5019c28e71-other.sqlite.bz2
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/repomd.xml
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/a27718cc28ec6d71432e0ef3e6da544b7f9d93f6bb7d0a55aacd592d03144b70-comps.xml
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/a7b004505b508e6ffba9021f7608a1b9eaa0c7224824641f4b8bd4084e145c5f-primary.xml.gz
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/a49d8df7475120084fd6415d528e1da744c7935ce95cb2b734c1b5276503c40a-other.xml.gz
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/eaab95137701c421b16286c054097fbfc6743f6038db2840f9eae1fd065439da-filelists.xml.gz
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/7a2b7d85c182a9335b488c156fe66265734e62c51c91cbc99122fb2ac67023e8-filelists.sqlite.bz2
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/4f32c0cc56ca03ad18cf40fd9a713d867daa5c04743f00a5c5cb5640235ecf89-primary.sqlite.bz2
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/repodata/cb6918cff2ac6d62023e9abc04fba6bc82ddebdadb01128e487f5db1386cce4e-updateinfo.xml.gz
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/openstack-packstack-14.0.1-0.20190814110430.d3eea20.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/subunit-filters-1.3.0-7.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/openstack-packstack-14.0.1-0.20200114160446.96fbb95.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/python3-junitxml-0.7-17.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/puppet-tempest-14.4.1-0.20190420122613.495927a.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/os/openstack-packstack-puppet-14.0.1-0.20190814110430.d3eea20.el8ost.noarch.rpm
./content/dist/layered/rhel8/x86_64/openstack-devtools/15/listing
./content/dist/layered/rhel8/x86_64/openstack-devtools/listing
./content/dist/layered/rhel8/x86_64/listing
./content/dist/layered/rhel8/listing
./content/dist/layered/listing
./content/dist/listing
./content/listing
```

also confirmed `generate-listing` works:
```
[vagrant@centos8-katello-devel 2022-07-13T20-40-44-00-00]$ hammer content-export generate-listing --task-id 0ecfcd81-8c77-453b-b788-021c49c695b5
Generated /var/lib/pulp/exports/Export_Org/SyncableView/1.0/2022-07-13T20-40-44-00-00
```

**Review by jeremylenz — APPROVED** [review]
APJ :+1:

**Comment by parthaa** [comment]
@jeremylenz need your approval here also.

**Pull request merged**